pages/index.js

```diff
@@ -1,25 +1,95 @@
 import useSWR from "swr";
+import { mutate } from "swr";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+import { useRef, useState } from "react";
+import Modal from "@/components/ui/Modal";
+
+import styled from "styled-components";
+
 import ChildCard from "@/components/ChildCard";
 import Spinner from "@/components/ui/Spinner";
-import { useRouter } from "next/router";
+
+const StyledLink = styled(Link)`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  position: relative;
+
+  width: 80%;
+  min-height: 150px;
+  padding: 10px;
+  overflow: hidden;
+  text-decoration: none;
+
+  border-radius: 20px;
+  background: #fff;
+  box-shadow: 4px 4px 4px 0px rgba(0, 0, 0, 0.1);
+`;
 
 export default function HomePage() {
   const router = useRouter();
+
   const { data: childrenData, isLoading } = useSWR("/api/children_items", {
     fallbackData: [],
   });
+  const [childId, setChildId] = useState(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  function openModal(id) {
+    setChildId((childId) => (childId = id));
```

Comment:
this can be simplified
```suggestion
    setChildId(childId);
```